### ISSUE TITLE
Closing Window while executing 'pause' when run from explorer results in long wait.

### ISSUE THREAD

**Issue by JamesG269** [PowerShell/PowerShell#9487]
# Steps to reproduce
Put 'pause' in a .ps1 script, execute from explorer 'open with pwsh.exe' (issue does not manifest if .ps1 file is run from the command line), while at pause prompt close the window by clicking the top 'X' to close the window instead of pressing a key.
```powershell

```

# Expected behavior
Window should close immediately after pause ask you to press a key and you hit the 'X', like it does when you press a key, there is probably some unintended issue here besides the unnecessary wait.
```none

```

# Actual behavior
Window waits several seconds then closes.
```none

```

# Environment data

Name                           Value
----                           -----
PSVersion                      6.2.0
PSEdition                      Core
GitCommitId                    6.2.0
OS                             Microsoft Windows 10.0.18362
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```none

```
I have seen this issue on multiple installs of Windows going back several versions of PowerShell.

**Comment by chuanjiao10** [comment]
this problem because powershell engine not “loading finished” or “initialization finished”.
Often appear multi-Progress powershell.exe script.
add "start-sleep -second 2" to your script ,then problem gone.
I don't think this is a problem with powershell.

**Comment by JamesG269** [comment]
I have pause at the end of long running scripts, and noticed this, and adding start-sleep did not do alleviate the issue.

**Comment by fmichaleczek** [comment]
@JamesG269 I try to reproduce your problem without success. 

Winversion : 10.0.18875
PSVersion : 6.2.0

File script.ps1 : 
```
Get-ChildItem $env:windir\System32 -Recurse
Get-ChildItem $env:windir\System32 -Recurse
pause
```
Does this script work as expect in your system ?
Could you temporary exclude the pwsh process and the powershell script from your Antivirus software and try again ?

**Comment by JamesG269** [comment]
Antivirus disabled temporarily through Gpedit.msc, still happened. I narrowed it down to something useful though, your post made me think to do more debugging, so I tried renaming my powershell profile so it would not load. And, the problem disappeared, through a process of elimination, I found the culprit lines:
```
Set-PSReadLineOption -Colors @{Parameter = "yellow"}
Set-PSReadLineOption -Colors @{type = "yellow"}
Set-PSReadLineOption -Colors @{string = "yellow"}
Set-PSReadLineOption -Colors @{operator = "yellow"}
Set-PSReadLineOption -Colors @{comment = "yellow"}
```
I use these, and only these to change console colors, and if any one of them is present in the powershell profile .ps1 file, the issue happens (again, only when run from explorer, not cmd line). Sorry I didn't think of it before, maybe there is a new way to change colors now? Though these seem to change the colors as expected..
